packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -213,7 +213,8 @@ class SharedPreferences {
       return MapEntry<String, dynamic>(newKey, value);
     });
     SharedPreferencesStorePlatform.instance =
-        InMemorySharedPreferencesStore.withData(newValues);
+        InMemorySharedPreferencesStore.withData(
+            newValues.cast<String, Object>());
```

Comment:
This cast isn't safe, since you are allowing `dynamic` in the API (meaning it won't do any type checking). You should change the API to `<String, Object>` instead of making a potentially invalid conversion at the end.